src/libraries/Native/Unix/System.Globalization.Native/pal_collation.c

```diff
@@ -839,11 +839,35 @@ int32_t GlobalizationNative_LastIndexOf(
 
     result = usearch_last(pSearch, &err);
 
-    // if the search was successful,
-    // we'll try to get the matched string length.
-    if (result != USEARCH_DONE && pMatchedLength != NULL)
+    // if the search was successful, we'll try to get the matched string length.
+    if (result != USEARCH_DONE)
     {
-        *pMatchedLength = usearch_getMatchedLength(pSearch);
+        int32_t matchLength = -1;
+
+        if (pMatchedLength != NULL)
+        {
+            matchLength = usearch_getMatchedLength(pSearch);
+            *pMatchedLength = matchLength;
+        }
+
+        // In case the search result pointing at last character (including Surrogate case) of the source string, we need to check if the target string
+        // constructed with characters which have no sort weights. The way we do that is to check the matched length is 0.
```

Comment:
```suggestion
        // In case the search result is pointing at the last character (including Surrogate case) of the source string, we need to check if the target string
        // was constructed with characters which have no sort weights. The way we do that is to check that the matched length is 0.
```